bot/services/positioin.py

```diff
@@ -0,0 +1,171 @@
+import urllib.parse
+
+from chromedriver_py import binary_path
+
+from selenium import webdriver
+from selenium.webdriver.chrome.service import Service
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.common.action_chains import ActionChains
+
+import time
+
+
+def prepare_url(search_phrase):
+    """Подготавливает поисковой запрос для браузера"""
+    url = urllib.parse.urljoin(
+        MAIN_URL,
+        '/catalog/0/search.aspx?search='
+        + search_phrase
+    )
+    return url
+
+
+def start_search(url):
+    """Открываем браузер с переданной страницей для начала поиска """
+    """и ждём (BROWSER_LOADING_TIME = 7) секунд для прогрузки страницы"""
+    browser.get(url)
+    time.sleep(BROWSER_LOADING_TIME)
+
+
+def get_full_page():
+    """Проматывает страницу в самый низ, ожидая (SCROLL_LOADING_TIME = 2)"""
+    """ секунды после каждой прокрутки для прогрузки скриптов"""
+    while True:
+        body = browser.find_element(By.TAG_NAME, "body")
+        actions = ActionChains(browser)
+        actions.move_to_element(body).send_keys(Keys.END).perform()
+        time.sleep(SCROLL_LOADING_TIME)
+        if browser.execute_script(
+            "return (window.innerHeight + window.scrollY) >= "
+            "document.body.scrollHeight;"
+         ):
+            break
+
+
+def palace_in_search(article):
```

Comment:
Немного еще потыкался, хочу узнать твое мнение, накидал вот такое решение позволяющее избавиться от ввода переменных result и my_good, а так же от конструкции трая
```suggestion
def palace_in_search(article):
    """Находит артикул на странице и возвращает его порядковый номер """
    """или None"""
    goods = browser.find_element(By.CLASS_NAME, 'product-card-list')
    goods_list = goods.find_elements(By.CSS_SELECTOR, 'article')
    articles = list(int(good.get_attribute('data-nm-id')) for good in goods_list)
    if article in articles:
        return goods_list.index(goods.find_element(By.ID, f'c{article}')) + 1
```